### PR TITLE
Build Nouveau on Windows

### PR DESCRIPTION
**Pull request by big-r81**
Porting Nouveau to build on Windows.

Commands:

```
.\make.cmd
.\make.cmd nouveau-test
.\make.cmd clean
```

The clean target ends with the following error:
```
PS C:\Users\couchdb\Documents\couchdb> .\make.cmd clean
==> snappy (clean)
==> bear (clean)
==> meck (clean)
==> folsom (clean)
==> hyper (clean)
==> ibrowse (clean)
==> jiffy (clean)
==> mochiweb (clean)
==> recon (clean)
==> proper (clean)
==> couch_epi (clean)
==> config (clean)
==> couch_log (clean)
==> khash (clean)
WARN:  Missing plugins: [pc]
==> b64url (clean)
WARN:  Missing plugins: [pc]
==> exxhash (clean)
==> ets_lru (clean)
==> chttpd (clean)
==> couch (clean)
==> couch_event (clean)
==> mem3 (clean)
==> couch_index (clean)
==> couch_mrview (clean)
==> couch_replicator (clean)
==> couch_pse_tests (clean)
==> couch_stats (clean)
==> couch_peruser (clean)
==> couch_tests (clean)
==> couch_dist (clean)
==> custodian (clean)
==> ddoc_cache (clean)
==> dreyfus (clean)
==> nouveau (clean)
==> fabric (clean)
==> global_changes (clean)
==> ioq (clean)
==> jwtf (clean)
==> ken (clean)
==> mango (clean)
==> rexi (clean)
==> setup (clean)
==> smoosh (clean)
==> weatherreport (clean)
==> couch_prometheus (clean)
==> rel (clean)
==> couchdb (clean)
The system cannot find the file specified.
make: [Makefile.win:424: clean] Error 2 (ignored)
The filename, directory name, or volume label syntax is incorrect.
make: [Makefile.win:426: clean] Error 123 (ignored)
The filename, directory name, or volume label syntax is incorrect.
make: [Makefile.win:427: clean] Error 123 (ignored)
Could Not Find C:\Users\couchdb\Documents\couchdb\src\*.dll
Could Not Find C:\Users\couchdb\Documents\couchdb\src\couch\priv\*.exe
The system cannot find the file specified.
make: [Makefile.win:431: clean] Error 2 (ignored)
The system cannot find the file specified.
make: [Makefile.win:432: clean] Error 2 (ignored)
dev\logs\nouveau.log - The process cannot access the file because it is being used by another process.
The system cannot find the file specified.
make: [Makefile.win:435: clean] Error 2 (ignored)
Could Not Find C:\Users\couchdb\Documents\couchdb\src\couch\priv\couch_js\config.h
The system cannot find the file specified.
make: [Makefile.win:437: clean] Error 1 (ignored)
> Task :clean FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':clean'.
> java.io.IOException: Unable to delete directory 'C:\Users\couchdb\Documents\couchdb\nouveau\build'
    Failed to delete some children. This might happen because a process has files open or has its working directory set in the target directory.
    - C:\Users\couchdb\Documents\couchdb\nouveau\build\libs\server-1.0-SNAPSHOT-dist.jar
    - C:\Users\couchdb\Documents\couchdb\nouveau\build\libs

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 5s
1 actionable task: 1 executed
make: *** [Makefile.win:438: clean] Error 1
```

It seems, nouveau isn't shutting down properly by our `dev/run` script. Needs additional investigation.

### CODE REVIEW

**Comment by big-r81** [comment]
Update:

A file handle to `dev/log/nouveau.log` is open too after running the test. The file can't be deleted for example.

**Comment by big-r81** [comment]
Fixed the open file handle issue by commit b7ca600e830e0ac6ccffbb75f92f4b3d71b0eacd.

Commands:
```
.\make.cmd
...
==> rel (compile)
==> couchdb (compile)
        1 file(s) copied.

BUILD SUCCESSFUL in 48s
```
```
.\make.cmd nouveau-test
...
Finished in 25.0 seconds (0.00s async, 25.0s sync)
22 tests, 0 failures

Randomized with seed 901471
```
```
.\make.cmd clean
...
BUILD SUCCESSFUL in 3s
1 actionable task: 1 executed
```

File: Makefile.win [hunk 1]
```diff
@@ -140,9 +140,11 @@ fauxton: share\www
 .PHONY: check
 # target: check - Test everything
 check: all
+	@$(MAKE) exunit
```
Comment (nickva) [inline Makefile.win:143]: I wonder if we have any exunit tests?

**Review by nickva — APPROVED** [review]
+1

The changes look good based on the diff. I didn't run it on Windows. It looks like you did and it worked, so I think that's good enough.

File: Makefile.win (hunk 1 above)
Comment (big-r81) [reply Makefile.win:143]: I don't know, I only made it at this part equal with our `Makefile`.

**Comment by big-r81** [comment]
Thanks Nick, if you have time, you can test it before merging.

File: Makefile.win (hunk 1 above)
Comment (nickva) [reply Makefile.win:143]: I figured it's just a main Makefile copy and it's not a blocker for PR. More of a general comment to cleanup it eventually both for Windows and in the main Makefile eventually.

**Review by nickva — APPROVED** [review]
@big-r81 since you insisted, as rightly you should have :-) I have tested the PR on the windows CI machine (on my own clone of couchdb). 

+1

Thanks keeping it updated

**Pull request merged**